controllers/cnbi/controller.go

```diff
@@ -303,37 +317,54 @@ func (r *CustomNBImageReconciler) ReconcilePipelineRun(name string, ctx *context
 			}
 			controllerutil.SetControllerReference(r.CNBi, pipelineRun, r.Scheme)
 
-			if err := r.Create(*ctx, pipelineRun); err != nil {
+			if err := r.Create(ctx, pipelineRun); err != nil {
 				logger.Error(err, "Unable to create PipelineRun")
-				updateStatus(metav1.ConditionTrue, "ErrorPipelineRunCreate", fmt.Sprintf("Unable to create PipelineRun. %s", err))
-				return err
+
+				setCondition(newStatus, meteorv1alpha1.ErrorPipelineRunCreate, metav1.ConditionTrue, "PipelineRunCreateFailed", err.Error())
+				return newStatus
 			}
-			updateStatus(metav1.ConditionTrue, "SuccessPipelineRunCreate", "Tekton PipelineRun was created.")
-			return nil
+			setCondition(newStatus, meteorv1alpha1.PipelineRunCreated, metav1.ConditionTrue, "PipelineRunCreated", fmt.Sprintf("%s PipelineRun created successfully", name))
+			return newStatus
 		}
-		logger.Error(err, "Error fetching PipelineRun")
 
-		updateStatus(metav1.ConditionFalse, "ErrorPipelineRun", fmt.Sprintf("Reconcile resulted in error. %s", err))
-		return err
+		logger.Error(err, "Error fetching PipelineRun")
+		setCondition(newStatus, meteorv1alpha1.GenericPipelineError, metav1.ConditionTrue, "PipelineRunGenericError", err.Error())
+		return newStatus
 	}
 
 	if len(pipelineRun.Status.Conditions) > 0 {
-		if len(pipelineRun.Status.Conditions) != 1 {
+		if len(pipelineRun.Status.Conditions) != 1 { // TODO observe tekton project if they stay with just one condition all the time
```

Comment:
Does that mean: we should add a feature watching for some behaviour of the tekton operator, *or* we should check later if the tekton project change the fact that they set one condition at a time ? I'm not sure from the formulation

Reply:
#techDebt